test/fixtures/full_config.yml

```diff
@@ -2355,6 +2357,7 @@ Style/CaseEquality:
   VersionAdded: '0.9'
   VersionChanged: '0.89'
   AllowOnConstant: true
+  AllowOnSelfClass: false
```

Comment:
- `AllowOnConstant`, which is enabled above, permits

    ```ruby
    SomeConstant === object
    ```

- `AllowOnSelfClass` if enabled, would permit

    ```ruby
    self.class === object
    ```